src/wlapplication.cc

```diff
@@ -503,30 +503,46 @@ void WLApplication::run() {
 			EditorInteractive::run_editor(
 			   EditorInteractive::Init::kLoadMapDirectly, filename_, script_to_run_);
 		}
-	} else if (game_type_ == GameType::kReplay) {
+	} else if (game_type_ == GameType::kReplay || game_type_ == GameType::kLoadGame) {
 		Widelands::Game game;
-
-		std::string map_theme, map_bg;
+		std::string title, message;
 		try {
-			game.create_loader_ui({"general_game"}, true, map_theme, map_bg);
-			game.set_ibase(new InteractiveSpectator(game, get_config_section()));
-			game.set_write_replay(false);
-			ReplayGameController rgc(game, filename_);
-			game.save_handler().set_allow_saving(false);
-			game.run(Widelands::Game::StartGameType::kSaveGame, "", true, "replay");
-		} catch (const std::exception& e) {
-			emergency_save(nullptr, game, e.what());
-		}
-	} else if (game_type_ == GameType::kLoadGame) {
-		Widelands::Game game;
-		game.set_ai_training_mode(get_config_bool("ai_training", false));
-		try {
-			game.run_load_game(filename_, script_to_run_);
+			if (game_type_ == GameType::kReplay) {
+				std::string map_theme, map_bg;
+				game.create_loader_ui({"general_game"}, true, map_theme, map_bg);
+				game.set_ibase(new InteractiveSpectator(game, get_config_section()));
+				game.set_write_replay(false);
+				ReplayGameController rgc(game, filename_);
+				game.save_handler().set_allow_saving(false);
+				game.run(Widelands::Game::StartGameType::kSaveGame, "", true, "replay");
+			} else {
+				game.set_ai_training_mode(get_config_bool("ai_training", false));
+				game.run_load_game(filename_, script_to_run_);
+			}
 		} catch (const Widelands::GameDataError& e) {
-			log_err("Game not loaded: Game data error: %s\n", e.what());
+			message = (boost::format(_("Widelands could not load the file \"%s\". The file format "
+			                           "seems to be incompatible.")) %
+			           filename_.c_str())
+			             .str();
+			message = message + "\n\n" + _("Error message:") + "\n" + e.what();
+			title = _("Game data error");
+		} catch (const FileNotFoundError& e) {
+			message =
+			   (boost::format(_("Widelands could not find the file \"%s\".")) % filename_.c_str())
+			      .str();
+			message = message + "\n\n" + _("Error message:") + "\n" + e.what();
+			title = _("File system error");
 		} catch (const std::exception& e) {
 			emergency_save(nullptr, game, e.what());
+			message = e.what();
+			title = _("Error message:");
+		}
+		g_sh->change_music("menu");
+		FsMenu::MainMenu m(true);
+		if (!message.empty()) {
+			m.show_messagebox(title, message);
```

Comment:
Please print the message to stdout as well (either here or in `show_messagebox()`)